core/src/main/java/com/google/zxing/aztec/decoder/Decoder.java

```diff
@@ -332,6 +333,22 @@ private static int readCode(boolean[] rawbits, int startIndex, int length) {
     return res;
   }
 
+  private static byte boolsToByte(final boolean[] array, final int start) {
+    byte b = 0;
+    for (int i = 0; i < 8; i++) {
+      if (array[start + i])
+        b |= 1 << (7 - i);
+    }
+    return b;
+  }
+
+  public static byte[] convertBoolArrayToByteArray(boolean[] boolArr) {
+    byte[] byteArr = new byte[boolArr.length / 8];
```

Comment:
Is it certain that the array's length will be a multiple of 8? otherwise this would not work.
Have you tested the result to see that it matches what you expect? Maybe piggy-backing a check for this result on some of the existing tests would be a good idea.
Finally, add braces for all if and for statcements.